core/src/main/scala/com/databricks/labs/remorph/parsers/snowflake/SnowflakeExpressionBuilder.scala

```diff
@@ -309,6 +310,33 @@ class SnowflakeExpressionBuilder()
       frame_spec = frameSpec)
   }
 
+  // see: https://docs.snowflake.com/en/sql-reference/functions-analytic#list-of-window-functions
+  private val rankRelatedWindowFunctions = Set(
+    "CUME_DIST",
+    "DENSE_RANK",
+    "FIRST_VALUE",
+    "LAG",
+    "LAST_VALUE",
+    "LEAD",
+    "NTH_VALUE",
+    "NTILE",
+    "PERCENT_RANK",
+    "RANK",
+    "ROW_NUMBER")
+
+  /**
+   * For rank-related window functions, snowflake's default frame deviate from ANSI standard. So in such case, we must
+   * make the frame specification explicit. see:
+   * https://docs.snowflake.com/en/sql-reference/functions-analytic#usage-notes-for-window-frames
+   */
+  private def snowflakeDefaultFrameSpec(windowFunction: ir.Expression): Option[ir.WindowFrame] = {
+    val rankRelatedDefaultFrameSpec = ir.WindowFrame(ir.RowsFrame, ir.UnboundedPreceding, ir.UnboundedFollowing)
+    windowFunction match {
+      case fn: ir.Fn if rankRelatedWindowFunctions.contains(fn.prettyName) => Some(rankRelatedDefaultFrameSpec)
```

Comment:
```suggestion
    windowFunction match {
      case fn: ir.Fn if rankRelatedWindowFunctions.contains(fn.prettyName) => 
        val rankRelatedDefaultFrameSpec = ir.WindowFrame(ir.RowsFrame, ir.UnboundedPreceding, ir.UnboundedFollowing)
        Some(rankRelatedDefaultFrameSpec)
```
forgotten to nit this on performance.